src/web/mjs/engine/Connector.mjs

```diff
@@ -423,19 +426,29 @@ export default class Connector {
      * and get all elements matching the given CSS selector.
      */
     async fetchDOM(request, selector, retries, encoding) {
-        retries = retries || 0;
+        retries = retries || this.retries;
         if(typeof request === 'string') {
             request = new Request(request, this.requestOptions);
         }
         // TODO: check if this will affect (replace) the input parameter?
         if(request instanceof URL) {
             request = new Request(request.href, this.requestOptions);
         }
-        const response = await fetch(request.clone());
-        if(response.status >= 500 && retries > 0) {
+
+        let response = null;
+        try {
```

Comment:
Do not use `try ... catch ...`, fail fast.
There are hundreds of system errors were a retry makes no sense (`OUT_OF_MEM`, `E_CONN_RESET`, `E_DISCONNECTED`, ...)